### ISSUE TITLE
SignalPlot using YOffset leaves dots at original position

### ISSUE THREAD

**Issue by raburton** [ScottPlot/ScottPlot#3086]
## Bug Report

**Issue:** SignalPlot using YOffset leaves dots at original position

**Reproducing:** Use Data.YOffset on a SignalPlot, and observe the dots left at the base position - see screenshot. Looks like the markers get left behind, except it's not the markers, because setting them to not visible doesn't help. However, if you try to zoom in on them, they will disappear at a certain level of zoom.

```cs
private double[] RandData() {
    double[] d = new double[10000];
    Random r = new Random();
    for (int i = 0; i < 10000; i++) d[i] = r.NextDouble()*100;
    return d;
}

private void Form1_Load(object sender, EventArgs e) {
    double[] data1 = new double[10000];
    double[] data2 = RandData();
    double[] data3 = RandData();
    
    Signal s1 = formsPlot1.Plot.Add.Signal(data1);
    s1.Marker.IsVisible = false;
    Signal s2 = formsPlot1.Plot.Add.Signal(data2);
    s2.Data.YOffset = 150;
    s2.Marker.IsVisible = false;
    Signal s3 = formsPlot1.Plot.Add.Signal(data3);
    s3.Data.YOffset = 300;
    s3.Marker.IsVisible = false;
}
```

![image](https://github.com/ScottPlot/ScottPlot/assets/433543/6a8a459a-de3b-48dd-ba68-eb19d5345755)

### System Details
* **ScottPlot Version:** 5.0.10-beta
* **Operating System:** Windows 10
* **Application Type:** WinForms
* **.NET Version:** .NET 8.0

**Comment by raburton** [comment]
More oddities with this, which might help identify the problem.

```cs

private void Form1_Load(object sender, EventArgs e) {

    double[] data1 = new double[10000];
    double[] data2 = new double[1];

    // uncomment this to produce the second screenshot
    //for (int i = 5000; i < 10000; i++) data1[i] = 100;

    Signal s1 = formsPlot1.Plot.Add.Signal(data1);
    s1.LineStyle.Color = Colors.Red;
    s1.LineStyle.Width = 5;
    s1.Data.YOffset = 100;

    // s2 just to keep the origin in frame
    Signal s2 = formsPlot1.Plot.Add.Signal(data2);
    s2.LineStyle.Color = Colors.Blue;
    s2.LineStyle.Width = 5;

}

```
With all zero data values, but YOffset of 100:
![image](https://github.com/ScottPlot/ScottPlot/assets/433543/27327812-2815-491a-8c4f-3102641f2c07)
Zoom in close enough and the lines disappears, it's made up only of the wrongly places dots.

With the second half set to 100 (uncomment the line in the code above), and YOffset of 100:
![image](https://github.com/ScottPlot/ScottPlot/assets/433543/0149230f-05fb-497d-9464-67fb7940a03c)
Zoom in to this one and even odder things happen...

**Comment by swharden** [comment]
Thanks for reporting this @raburton! 

I suspect it's been fixed in #2949 because your code looks good on my machine now. These updates should be in the next release which I plan to push to NuGet within the next few days 👍 

![image](https://github.com/ScottPlot/ScottPlot/assets/4165489/68a0a960-f273-4be6-94b0-52fe3969bbaa)

![image](https://github.com/ScottPlot/ScottPlot/assets/4165489/bacbfe98-aa6d-4686-a186-6d8f6f69b5f7)
